### DISCUSSION TITLE
Redesign of the Queries tab

### DISCUSSION THREAD

**Discussion by onmax** [supabase/supabase#16874] [Feature Requests]
Hello Team! 👋 

I would like to suggest two new improvements to the Supabase SQL Queries view:

- **Folder structure of queries**: In my current project I use about 10 queries and it looks a bit messy, it would be nice to be able to put the queries in different folders.
- **Select the queries you want to see**: Instead of putting all the queries in one column: 'Templates', 'Quickstarts', 'Shared Queries' and 'Your Queries', it would be nice to have a select to change them like we have in the Tables view for example.

I did a quick design which you can check out in this [figma file](https://www.figma.com/file/QuRdbfhsTbLp4vJ9U50iy1/Untitled?type=design&node-id=0%3A1&mode=design&t=YhDKpsApjgRmcaPl-1).

## Comparassion

<table>
<thead>
<tr>
      <td><h3>Old design</h3></td>
      <td><h3>Proposed design</h3></td>
</tr>
</thead>
  <tbody>
    <tr>
      <td>
<img src="https://github.com/supabase/supabase/assets/22072217/9f79e6e8-8cc3-470f-bc6b-a6925d22792e" />
</td>
      <td>
<img src="https://github.com/supabase/supabase/assets/22072217/7c20043d-075a-4e9d-9888-d6ff87a275a9" />
</td>
    </tr>

  </tbody>
</table>

## 🙋‍♂️ Pitching in
I'd love to collaborate and bring this vision to life and contribute wherever possible.

I'd love to hear everyone's thoughts! Let's make Supabase even better! 🚀

Cheers, Maxi.

Thanks for making Supabase! 💚💚

**Comment by saltcod** [comment]
Hey @onmax great! Thanks for this. 

This one came up quite a bit in the SQL Editor rfc we posted a while back, but we haven't gotten to it yet. 
https://github.com/orgs/supabase/discussions/14206

We'd love to have you contribute! I think the last we spoke about this internally, we were thinking of using the same paradigm that storage files and buckets use. 

_I think_ to put a file in a folder called images/pets/cats, you actually bake all of that into the filename. 
So the file whos-a-good-boy.png would be called `images/pets/cats/whos-a-good-boy.png`. 

Been a bit since I've looked at it, but this could be a quick way to do it. Happy to provide any guidance or feedback if you get stuck into this one!

**Reply by onmax** [reply]
Sorry, for the delay, just saw your message.

I don't fully understand your idea. Do you want to have each folder in its own column?

![image](https://github.com/supabase/supabase/assets/22072217/aacf838a-0a58-4932-a55f-a643d3acd0ca)

In that case, it will reduce considerably the space for the editor itself. Did I misunderstand you?
